### ISSUE TITLE
Draft 6 support

### ISSUE THREAD

**Issue by yuloh** [thephpleague/json-guard#114]
This is already supported by the dereferencer. 

We need a new rule set and constraints.

**Comment by handrews** [comment]
draft-06 is out!

[Guide for core and validation](https://github.com/json-schema-org/json-schema-spec/wiki/FAQ:-draft-wright-json-schema%5B-validation%5D-01)
[Guide for hyper-schema](https://github.com/json-schema-org/json-schema-spec/wiki/FAQ:-draft-wright-json-schema-hyperschema-01)

**Comment by danballance** [comment]
Hi, just reading this issue through. Does this mean I can validate (pretty simple) Draft 6 schemas with json-guard? PHP support for draft 6 looks non-existent atm and I need draft 6 for the "named enum" work aropund described here:

https://github.com/json-schema-org/json-schema-spec/issues/57

This issue looks promising. Anything I can do to help if this isn't already working out of the box?

**Comment by handrews** [comment]
@danballance in the meantime, you can always use a single-element `enum` in place of `const` as a further workaround.  `{"const": 42}` is just shorthand for `{"enum": [42]}`.  It's easier to read, and easier for tools to recognize as a special case for code or UI generation, but the validation effect is the same.

**Comment by yuloh** [comment]
Hello,

> Hi, just reading this issue through. Does this mean I can validate (pretty simple) Draft 6 schemas with json-guard? 

Not yet.  [json-reference](http://json-reference.thephpleague.com/) will mostly work with Draft 6 schemas (it supports `$id`) but this library doesn't support it yet.

> Anything I can do to help if this isn't already working out of the box?

To support draft 6 we need a new [rule set](https://github.com/thephpleague/json-guard/tree/master/src/RuleSet) (see the docs for more info) and [constraints](https://github.com/thephpleague/json-guard/tree/master/src/Constraint/DraftFour) for the changes in Draft 6.  Most of the constraints can probably decorate the draft 4 version to add the new logic.  The JSON Schema Test Suite will need to be updated to pull in the draft 6 tests (see composer.json) and you can use that to write the new rule set.

If you or someone else wants to tackle that let me know and I can answer any questions.  I probably won't have time to do this myself anytime soon.  I am planning on tackling #122 first since I need it for work.

**Comment by danballance** [comment]
@handrews yes that's what we are doing for now. Although I am seeing some other features in draft 6 that we could make use of.

@yuloh thanks for clarifying. I am considering trying to take this on as a weekend project, but I'm not free now until January 2018. I really would like to help with this. I'll see what I can organise in my home life and come back to you.

**Comment by handrews** [comment]
@danballance @yuloh Note that [draft-07 is now nearing publication](http://json-schema.org/work-in-progress)

**Comment by philsturgeon** [comment]
@yuloh ping Draft 7 is out!
